harmony-api/prisma/schema.prisma

```diff
@@ -0,0 +1,168 @@
+  Post    Post[]
+}
+
+model PollOption {
+  post_id Int    @id
+  option  String
+  poll    Poll   @relation(fields: [post_id], references: [post_id])
+}
+
+model Discussion {
+  post_id Int    @id @map("post_id")
+  content String
+  Post    Post[]
+}
+
+model Review {
+  post_id Int     @id @map("post_id")
+  stars   Decimal
+  content String
+  Post    Post[]
+}
+
+model Follow {
+  following_id Int
+  follower_id  Int
+  follow_time  DateTime
+  following    User     @relation("Following", fields: [following_id], references: [user_id])
+  follower     User     @relation("Follower", fields: [follower_id], references: [user_id])
+
+  @@id([following_id, follower_id])
+}
+
+model Like {
+  user_id Int
+  post_id Int
+  user    User @relation(fields: [user_id], references: [user_id])
+  post    Post @relation(fields: [post_id], references: [post_id])
+
+  @@id([user_id, post_id])
+}
+
+model Comment {
+  comment_id Int     @id @default(autoincrement())
+  user_id    Int
+  post_id    Int
+  content    String?
+  user       User    @relation(fields: [user_id], references: [user_id])
+  post       Post    @relation(fields: [post_id], references: [post_id])
+}
+
+model Album {
+  album_id          Int           @id @default(autoincrement())
+  album_name        String        @unique
+  album_description String?
+  artists           Artist[]
+  songs             Song[]
+  Publish           Publish[]
+  RelateAlbum       RelateAlbum[]
+}
+
+model Publish {
+  album_id  Int    @id
+  artist_id Int
+  album     Album  @relation(fields: [album_id], references: [album_id])
+  artist    Artist @relation(fields: [artist_id], references: [artist_id])
+}
+
+model Song {
+  song_id          Int          @id @default(autoincrement())
+  artist_id        Int
+  song_name        String
+  song_description String?
+  release_time     DateTime?
+  album_id         Int?
+  artist           Artist       @relation(fields: [artist_id], references: [artist_id])
+  album            Album?       @relation(fields: [album_id], references: [album_id])
+  RelateSong       RelateSong[]
+}
+
+// The RelateTo model is supposed to function like an abstract model
+model RelateTo {
+  post_id Int  @id @map("post_id")
+  post    Post @relation(fields: [post_id], references: [post_id])
+}
+
+model RelateSong {
+  song_id Int
+  post_id Int
+  song    Song @relation(fields: [song_id], references: [song_id])
+  post    Post @relation(fields: [post_id], references: [post_id])
+
+  @@id([post_id, song_id])
+}
+
+model RelateAlbum {
+  album_id Int
+  post_id  Int
+  album    Album @relation(fields: [album_id], references: [album_id])
+  post     Post  @relation(fields: [post_id], references: [post_id])
+
+  @@id([post_id, album_id])
+}
+
+model RelateArtist {
+  artist_id Int
+  post_id   Int
+  artist    Artist @relation(fields: [artist_id], references: [artist_id])
+  post      Post   @relation(fields: [post_id], references: [post_id])
+
+  @@id([post_id, artist_id])
```

Comment:
Since each post relates to exactly one song/album/artist, could we just condense this all into the Post to avoid as many joins? 
i.e.
```
model Post {
  postId           Int            @id @default(autoincrement()) @map("post_id")
  userId           Int            @map("user_id")
  title            String
  ...
  songId           Int            @map("song_id")
  song             Song?          @relation(fields: [songId], references: [songId])
  // and so on for artist and album
}
```

Reply:
I kinda agree with you on that. The whole purpose of the "RelateTo" model was to just act as a sort of work-around for querying with SQL. Since Prisma can handle the whole relation thing, I see why this would work fine. See (d156137)